src/components/product/product.service.ts

```diff
@@ -381,18 +345,18 @@ export class ProductService {
       .return('p')
       .first();
 
-    if (!produces) {
+    if (!pr) {
       const scriptureReferences = await this.listScriptureReferences(
-        baseNodeProps.id,
+        id,
         'Product',
         session
       );
+
       return {
```

Comment:
This could be done once instead of both here and below. (Other than the scriptureReferences prop)